cmdstanpy/utils.py

```diff
@@ -234,7 +236,18 @@ def cmdstan_version_before(major: int, minor: int) -> bool:
 
     :return: True if version at or above major.minor, else False.
     """
-    cur_version = cmdstan_version()
+    cur_version = None
+    if info is None:
+        cur_version = cmdstan_version()
+    else:
+        if (
+            info['stan_version_major'] is not None
+            and info['stan_version_minor'] is not None
+        ):
+            cur_version = (
+                int(info['stan_version_major']),
+                int(info['stan_version_minor']),
+            )
```

Comment:
This will fail to get any version if info_dict is `{}`, which is a possibility. I guess in that case we know the cmdstan version is less than we're asking of it?

Reply:
The way I would change it is `info is None or 'stan_version_major' not in info`